lib/core/models/repositories/roleRepository.js

```diff
@@ -181,25 +194,130 @@ class RoleRepository extends Repository {
    * @returns Promise
    */
   validateAndSaveRole (role, options) {
-    return role.validateDefinition()
+
+    return role
+      .validateDefinition()
       .then(() => {
         if (role._id === 'anonymous' && !role.canLogIn()) {
           errorsManager.throw('security', 'role', 'login_required');
         }
 
-        this.kuzzle.emit(
-          'core:roleRepository:save',
-          { _id: role._id, controllers: role.controllers });
-
-        return this.persistToDatabase(role, options);
+        this.kuzzle.emit('core:roleRepository:save', {
+          _id: role._id,
+          controllers: role.controllers
+        });
+      })
+      .then(() => this.checkRoleNativeRights(role))
+      .then(() => {
+        const force = options ? options.force : false;
+        this.checkRolePluginsRights(role, force);
       })
+      .then(() => this.persistToDatabase(role, options))
       .then(() => this.loadOneFromDatabase(role._id))
       .then(updatedRole => {
         this.roles.set(role._id, updatedRole);
         return updatedRole;
       });
   }
 
+  /**
+   * Given a Role object, checks if its controllers and actions exist.
+   *
+   * @param {Role} role
+   */
+  checkRoleNativeRights(role) {
+    const roleRightsError = errorsManager.wrap('security', 'role');
+    Object.keys(role.controllers).forEach(roleController => {
+      if (roleController !== '*' && !this.kuzzle.funnel.controllers.has(roleController)) {
+        if (roleController.split('/').length > 1) {
+          // Meaning this is a plugin controller. So we skip it.
+          return;
+        }
+        roleRightsError.throw('unknown_controller', role._id, roleController);
+      }
+      if (roleController === '*') {
+        this.kuzzle.funnel.controllers.forEach(controller => {
+          Object.keys(role.controllers['*'].actions).forEach(action => {
+            if (action !== '*' && !controller._isAction(action)) {
+              roleRightsError.throw('unknown_controller_action', role._id, action, controller);
+            }
+          });
+        });
+      }
+      else {
+        const controller = this.kuzzle.funnel.controllers.get(roleController);
+        const actions = Object.keys(role.controllers[roleController].actions); 
+        actions.forEach(action => {
+          if (action !== '*' && !controller._isAction(action)) {
+            roleRightsError.throw('unknown_controller_action', role._id, action, roleController);
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Given a Role object, checks if its controllers and actions exist in plugins.
+   *
+   * @param {Role} role
+   * @param {Force} force
+   */
+  checkRolePluginsRights(role, force = false) {
```

Comment:
Use named parameters syntax instead for readability
```suggestion
  checkRolePluginsRights(role, { force=false } = {}) {
```
So you can call it like this 
```
checkRolePluginsRights(role, { force: true })
checkRolePluginsRights(role) // force: false by default
```